packages/remark-wiki-link/src/lib/fromMarkdown.ts

```diff
@@ -79,7 +79,7 @@ function fromMarkdown(opts: FromMarkdownOptions = {}) {
       data: { isEmbed, target, alias },
     } = wikiLink;
     // eslint-disable-next-line no-useless-escape
-    const wikiLinkWithHeadingPattern = /([\p{Letter}\d\s\/\.-_]*)(#.*)?/u;
+    const wikiLinkWithHeadingPattern = /([\p{Letter}\d\s\/\.-_-]*)(#.*)?/u;
```

Comment:
Haven't noticed it before, but `.-_` denotes a range between `.` and `_` unicode characters, and it seems it indeed doesn't include `-`. And while adding `-` at the end like you did will surely solve the problem, I think we shouldn't have included this range (`.-_`) preceding it in the first place. The range will include all the characters between the period (U+002E) and the underscore (U+005F), e.g. `>`, `<`, `?`, `;`, which are not valid characters in file names. Rather than using this broad range, we should be very explicit here. Thus, let's replace the original pattern with:

```suggestion
    const wikiLinkWithHeadingPattern = /([\p{Letter}\d\s\/\.\-_]*)(#.*)?/u;
```

Please note, this pattern should also be changed [here](https://github.com/datopian/portaljs/blob/3366086d87e763d2f0d342a5f7bd9df4329a4d45/packages/remark-wiki-link/src/lib/html.ts#L67C20-L67C20). It seems we forgot about it last time.

Reply:
Yep, I noticed but I didn't change it because I thought that was a proposal range, and in the HTML was that same thought (to change something planned before). Thank you for all the clarification and help. I'll fix it.